### PR TITLE
docs(rich-text-parser): add draft feature files

### PR DESCRIPTION
**Pull request by HendrikThePendric**
This is equivalent to this https://github.com/dhis2/ui-widgets/pull/289 one, which I closed because we are archiving `@dhis2/ui-core`.

### CODE REVIEW

**Comment by HendrikThePendric** [comment]
Closing this one in favour of https://github.com/dhis2/ui-widgets/pull/289

**Comment by HendrikThePendric** [comment]
Sorry, I closed the wrong PR 🤦 . I've now reopened this one and closed https://github.com/dhis2/ui-widgets/pull/289

File: cypress/drafts/RichTextParser/Emoticons.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Feature: Emoticon parsing
+
+  Scenario Outline: Emoticons strings
+      Given there is a text string of <string>
+      And the string is not preceded immediately by a non-space character
```
Comment (Mohammer5) [inline cypress/drafts/RichTextParser/Emoticons.feature:5]: I don't like this rule. What if I want to have an emoticon at the end of a sentence, or at the end of brackets.

**Comment by HendrikThePendric** [comment]
I've removed myself from the list of assignees and have added @cooper-joe. This seemed more appropriate since he wrote the initial version of this PR. I only moved it to the mono-repo.

File: cypress/drafts/RichTextParser/Emoticons.feature (hunk 1 above)
Comment (cooper-joe) [reply cypress/drafts/RichTextParser/Emoticons.feature:5]: I think the common workaround is to add a space to trigger the emoticon replacement and then backspace and add the extra bracket, comma, etc. Although slightly irritating, I think this is preferable. Interpreting `:)` anywhere as an emoticon would, I think, result in more unintended consequences.

**Comment by cooper-joe** [comment]
> I can see why we would want feature files here, on the other hand, these look like perfect jest candidates, there's no rendering involved, it's pure input->output, while feature files should cover user interaction.
> 
> So I don't think we'd need feature files for this, but rather for something like "given the user highlighted some text. When the user clicks the bold button, then selected text should be bold and not highlighted anymore"

@Mohammer5, I have no opinion about feature files vs. jest, however it would be good to get this PR merged or closed. What do you think would be the best way to move forward here?

**Comment by Mohammer5** [comment]
I think here we have to differentiate between behavior (cypress) and parsing (jest). I think the feature files of this PR are about parsing only, so we shouldn't use cypress.

It's nice to have the feature files though, so maybe we should consider jest-cucumber for the `ui` repository (cc @ismay @amcgee @varl @HendrikThePendric) as these specs are written by @cooper-joe and it's very nice/comfy to just get a list of feature files instead of having to parse a jira issue or google docs file.

**Comment by ismay** [comment]
> I think here we have to differentiate between behavior (cypress) and parsing (jest). I think the feature files of this PR are about parsing only, so we shouldn't use cypress.
> 
> It's nice to have the feature files though, so maybe we should consider jest-cucumber for the `ui` repository (cc @ismay @amcgee @varl @HendrikThePendric) as these specs are written by @cooper-joe and it's very nice/comfy to just get a list of feature files instead of having to parse a jira issue or google docs file.

Personally I would just use jest without cucumber. I think the indirection and added maintenance that jest-cucumber adds isn't worth the improved readability of the specs for me. In my opinion when written properly, jest tests by themselves are already quite readable. Enough that I'd say they don't need to be improved further.

**Comment by HendrikThePendric** [comment]
This presents us with a bit of a dilemma:
- I find that these feature files hold valuable information so they should not be discarded
- On the other had I also agree that these type of tests are unit tests and probably don't need feature files

So ultimately what should happen is that a set of unit tests is written that covers everything that's described in these feature files. But the most sensible time to write these is when we get to work on the component, which I don't think will be very soon....

So perhaps for now we do this:
- Add plentiful comments to these feature files that explain that these feature files should probably be converted to jest test when working on the component (and possibly look into using `jest-cucumber`, I'm not a huge fan).
- Just merge these feature files to the drafts folder

**Comment by Mohammer5** [comment]
@cooper-joe would you be able to write test file skeletons? I mean not the actual test code itself, but rather like this:

```js
describe('Transfer - Highlight range of option', () => {
  it('should select the option after shift clicking the option when not selected', () => {})
  it('should deselect the option after shift clicking the option when selected', () => {})
  // ...
})
```

**Comment by cooper-joe** [comment]
> @cooper-joe would you be able to write test file skeletons and probably understand test files when reading through them when they have actual test code? I mean not the actual test code itself, but rather like this:
> 
> ```js
> describe('Transfer - Highlight range of option', () => {
>   it('should select the option after shift clicking the option when not selected', () => {})
>   it('should deselect the option after shift clicking the option when selected', () => {})
>   // ...
> })
> ```

I'm sure I could learn to do so! Although I do think that the behaviour is easier to define when writing feature files than the above example. If it would simplify this process then I am open to giving it a try.

**Comment by Mohammer5** [comment]
My preference would be to use feature files, but I'm outnumbered :wink: We could schedule a meeting and I can give you an introduction to jest test files

**Comment by ismay** [comment]
> My preference would be to use feature files, but I'm outnumbered

I know it's in jest (💯 pun credits for me), but I wouldn't want to force a decision just through numbers. I would prefer to go with the best solution, even if the majority disagrees. I wouldn't mind taking a deeper look and weighing the pros/cons, there could be things I'm overlooking.

**Comment by HendrikThePendric** [comment]
Closing this due to inactivity

**Pull request closed without merging**